src/Evolution/Systems/ScalarWave/Initialize.hpp

```diff
@@ -0,0 +1,150 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <array>
+#include <cstddef>
+#include <tuple>
+#include <utility>  // IWYU pragma: keep
+#include <vector>
+
+#include "DataStructures/DataBox/DataBox.hpp"
+#include "DataStructures/DataBox/DataBoxTag.hpp"
+#include "DataStructures/DataBox/Prefixes.hpp"
+#include "DataStructures/Variables.hpp"  // IWYU pragma: keep
+#include "Domain/Mesh.hpp"
+#include "Domain/Tags.hpp"
+#include "Evolution/Initialization/DiscontinuousGalerkin.hpp"
+#include "Evolution/Initialization/Domain.hpp"
+#include "Evolution/Initialization/Evolution.hpp"
+#include "Evolution/Initialization/Limiter.hpp"
+#include "Evolution/Initialization/NonConservativeInterface.hpp"
+#include "Evolution/Systems/ScalarWave/Tags.hpp"
+#include "NumericalAlgorithms/LinearOperators/Divergence.tpp"
+#include "Parallel/ConstGlobalCache.hpp"
+#include "PointwiseFunctions/AnalyticData/Tags.hpp"
+#include "PointwiseFunctions/AnalyticSolutions/Tags.hpp"
+#include "PointwiseFunctions/GeneralRelativity/IndexManipulation.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Tags.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/MakeWithValue.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace ScalarWave {
+namespace Actions {
+namespace detail {
```

Comment:
Since almost all of the code in this file is a duplicate of what is in GRMHD, it should be factored out and generalized instead.

Reply:
I agree in principle, but I wasn't sure how to do this in practice. I could use some more detailed advice about what to factor and how to do it. 

Some questions I have so far:

1. Where should the common part of the code live, that the specific parts include?
 
2. How would you factor the overlapping parts of the code out? ScalarWave still needs its own initialize action, right? If so, then how do you factor out common using statements, tags, etc.?

Reply:
Would writing a generic initialize action struct in Evolution/Initialization, and then inheriting from it, filling in the specific parts be a good approach to making this generic?